Python Tutorial Exercises(Week 3)/#2.py

```diff
@@ -0,0 +1,76 @@
+'''
+This script prints a list of all prime numbers found between a given range of x and y inclusively. 
+This script includes a while loop and a for loop.
+The outer loop tells us all the numbers in the given range and 
+the inner loop helps us find all prime numbers.
+Create a new list 'z' to store the values we find from the inner loop.
+'''
+x = 20
+y = 47
+prime_number_list = []
+u = x
+
+# for loop
+for i in range(x,y+1):
+    for a in range(2,i):
+        if i%a == 0:
+            break
+    else:
+        prime_number_list.append(i)
+print(prime_number_list)
+
+#while loop
+# set range
+while u >= x and u <= y:
+    count = 0
+    #q = 2 is used for checking the factor of the number
+    # we start dividing by 2. 
+    # This is because divide 0 is invalid and any number can be divided by 1.
+    # So we will make a variable with the starting value of 2.
+    q = 2
+    
+    #use a while loop to calculate the prime number. 
+    # The smallest prime number is 2. the largest factor for q must be 
+    # smaller and equal to u/2, and check all integers smaller or equal 
+    # or smaller and equal to u/2. 
+    while q <= u/2:
+        #We are dividing the number by all the numbers using u%q == 0
+        if u%q == 0:
+            count += 1      
+            #The break statement is used to come out of 
+            #the loop as soon we get any positive divisor then no further check is required.
+            break
+        q += 1
+    #check the count number
+    if count == 0:
+        prime_number_list.append(u)
+    u += 1
+#printing the prime numbers between x and y
+print(prime_number_list)
+
+Min = 20
+Max = 47
+prime_numbers = []
+i = Min
+
+while i >= Min and i <= Max:
```

Comment:
I'm confused as to why there are two while loop solutions...

Reply:
You asked me can I think of a way that doesn't require that count variable. The second doesn't require that count variable.

Reply:
Do you want me to remove the first while loop?

Reply:
Yes, remove one or the other. Having either the `count` and `isPrime` variables achieve the same thing, they both act as a tracking variable which isn't needed as you can achieve this with just `while/else`. Similar to how the for loop solution doesn't need a tracking variable.

Reply:
You haven't addressed everything from this previous comment above, specifically this part "You have also removed all the useful comments from the previous while loop."